tsdb/index/inmem/inmem.go

```diff
@@ -490,20 +493,29 @@ func (i *Index) measurementNamesByNameFilter(op influxql.Token, val string, rege
 			matched = !regex.MatchString(m.Name)
 		}
 
-		if !matched {
-			continue
+		if matched && m.Authorized(auth) {
+			names = append(names, m.name)
 		}
-		names = append(names, []byte(m.Name))
 	}
 	bytesutil.Sort(names)
 	return names
 }
 
 // measurementNamesByTagFilters returns the sorted measurements matching the filters on tag values.
-func (i *Index) measurementNamesByTagFilters(filter *TagFilter) [][]byte {
+func (i *Index) measurementNamesByTagFilters(auth query.Authorizer, filter *TagFilter) [][]byte {
 	// Build a list of measurements matching the filters.
 	var names [][]byte
 	var tagMatch bool
+	var authorized bool
+
+	// valEqual determins if the provided []byte is equal to the tag value
+	// to be filtered on.
+	valEqual := func(b []byte) bool {
+		if filter.Op == influxql.EQ || filter.Op == influxql.NEQ {
+			return bytes.Equal([]byte(filter.Value), b)
+		}
+		return filter.Regex.Match(b)
+	}
```

Comment:
Does this need to be an inline function? Can you move it out and just pass `filter` in?